unstructured/metrics/element_type.py

```diff
@@ -20,3 +20,52 @@ def get_element_type_frequency(
         else:
             frequency[key] += 1
     return frequency
+
+
+def calculate_element_type_percent_match(
+    output: Dict,
+    source: Dict,
+    category_depth_weight: float = 0.5,
+) -> float:
+    """
+    Calculate the percent match between two frequency dictionary. Intended to use with
+    `get_element_type_frequency` function. The function counts the absolute exact match
+    (type and depth), and counts the weighted match (correct type but different depth),
+    then normalized with source's total elements.
+    """
+    if len(output) == 0 or len(source) == 0:
+        return 0.0
+
+    output_copy = output.copy()
+    source_copy = source.copy()
+    total_source_element_count = 0
+    total_match_element_count = 0
+
+    unmatched_depth_output = {}
+    unmatched_depth_source = {}
+
+    for k, v in output_copy.items():
+        if k in source_copy:
+            match_count = min(output_copy[k], source_copy[k])
+            total_match_element_count += match_count
+            total_source_element_count += match_count
```

Comment:
should this be source_count and not match_count?

Reply:
It'll continue counting the rest of the source later since we're looping through the output list now so we can only track the matched one in this loop.